src/main/java/Ui.java

```diff
@@ -0,0 +1,100 @@
+import java.util.Scanner;
+
+/**
+ * Represents the interaction between Duke and the User. Both in reading the input and displaying messages / errors
+ * 		of the program
+ */
+public class Ui {
+	private Scanner scanner;
+	private StringBuilder outputHolder;
+
+	/**
+	 * Default constructor for UI, initializes a scanner object to read user inputs.
+	 */
+	public Ui() {
+		scanner = new Scanner(System.in);
+		outputHolder = new StringBuilder();
+	}
+
+	/**
+	 * show errors from attempting to load tasks from the archival file
+	 * @param ex exception thrown from loading tasks from the archival file
+	 */
+	public void showLoadingErrors(Exception ex) {
+		this.outputHolder.append(ex.getMessage());
+	}
+
+	/**
+	 * show the welcome message to the user
+	 */
+	public void showWelcome() {
+		this.outputHolder.append("Welcome to Duke. What can I do for you today?");
+	}
+
+	/**
+	 * Displays messages to the user
+	 * @param toShow the string to be displayed to the user
+	 */
+	public void displayLine(String toShow) {
+		this.outputHolder.append(toShow);
+	}
+
+	/**
+	 * simply display a break line to organise the outputs
+	 */
+	public void showBreakLine() {
```

Comment:
As you've completed the GUI, consider either removing the `Ui` class outright (as it is dead code) or refactoring it to handle the UI elements of the application.